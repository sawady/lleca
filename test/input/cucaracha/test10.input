
fun main() {
  x := 104
  y := 111
  putChar(x)
  putChar(y)
  x := 108
  y := 97
  putChar(x)
  putChar(y)
  x := 10
  putChar(x)
}
