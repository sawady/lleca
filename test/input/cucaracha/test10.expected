Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtAssign(
        x,
        ExprConstNum(
          104
        )
      ),
      Cons(
        StmtAssign(
          y,
          ExprConstNum(
            111
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprVar(
                x
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprVar(
                  y
                ),
                Nil
              )
            ),
            Cons(
              StmtAssign(
                x,
                ExprConstNum(
                  108
                )
              ),
              Cons(
                StmtAssign(
                  y,
                  ExprConstNum(
                    97
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprVar(
                        x
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprVar(
                          y
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtAssign(
                        x,
                        ExprConstNum(
                          10
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprVar(
                              x
                            ),
                            Nil
                          )
                        ),
                        Nil
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Nil
)
